CMakeModules/build_clBLAS.cmake

```diff
@@ -0,0 +1,39 @@
+include(ExternalProject)
+
+set(prefix ${CMAKE_BINARY_DIR}/third_party/clBLAS)
+set(clBLAS_location ${prefix}/lib/import/${CMAKE_STATIC_LIBRARY_PREFIX}clBLAS${CMAKE_STATIC_LIBRARY_SUFFIX})
+if(CMAKE_VERSION VERSION_LESS 3.2 AND CMAKE_GENERATOR MATCHES "Ninja")
+  message(WARNING "Building clBLAS with Ninja has known issues with CMake older than 3.2")
+  set(byproducts)
+else()
+  set(byproducts BYPRODUCTS ${clBLAS_location})
+endif()
+
+ExternalProject_Add(
+  clBLAS-external
+  GIT_REPOSITORY https://github.com/clMathLibraries/clBLAS.git
+  GIT_TAG 47662a6ac1186c756508109d7fef8827efab4504
```

Comment:
I don't know if the tag is necessary, but it is a good idea to keep it in my opinion: it ensures that the version used is the expected one at the time the code was written, a bit like having clBLAS and clFFT directly in the repository or in a submodule.